packages/react-native/Libraries/StyleSheet/processBackgroundImage.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @format
+ * @flow
+ */
+
+'use strict';
+
+import type {____ColorValue_Internal} from './StyleSheetTypes';
+
+const processColor = require('./processColor').default;
+
+const LINEAR_GRADIENT_REGEX = /linear-gradient\s*\(((?:\([^)]*\)|[^())])*)\)/g;
+const COLOR_STOP_REGEX =
+  /\s*((?:(?:rgba?|hsla?)\s*\([^)]+\))|#[0-9a-fA-F]+|[a-zA-Z]+)(?:\s+([0-9.]+%?))?\s*/g;
+const DIRECTION_REGEX =
+  /^to\s+(?:top|bottom|left|right)(?:\s+(?:top|bottom|left|right))?/;
+const ANGLE_UNIT_REGEX = /^.*?(deg|grad|rad|turn)$/;
+const TRANSPARENT = 0; // rgba(0, 0, 0, 0)
+const TO_BOTTOM_START_END_POINTS = {
+  start: {x: 0.5, y: 0},
+  end: {x: 0.5, y: 1},
+};
+
+export type BackgroundImagePrimitive = {
+  type: 'linearGradient',
+  start: {x: number, y: number},
+  end: {x: number, y: number},
+  colorStops: $ReadOnlyArray<{
+    color: ____ColorValue_Internal,
+    position: number,
+  }>,
+};
+
+export default function processBackgroundImage(
+  backgroundImage: $ReadOnlyArray<BackgroundImagePrimitive> | string,
+): $ReadOnlyArray<BackgroundImagePrimitive> {
```

Comment:
The output has some more guarantees that the input doesn't. E.g. output of processor `ColorValue` should become `ProcessedColorValue`, right?